src/rgw/rgw_http_client.cc

```diff
@@ -337,6 +337,10 @@ int RGWHTTPClient::init_request(rgw_http_req_data *_req_data, bool send_data_hin
   }
   if (has_send_len) {
     curl_easy_setopt(easy_handle, CURLOPT_INFILESIZE, (void *)send_len); 
+    h = curl_slist_append(h, "Transfer-Encoding:");
+    stringstream ss;
+    ss << "Content-Length: " << send_len;
+    h = curl_slist_append(h, ss.str().c_str());
```

Comment:
is all of this needed? shouldn't setting `CURLOPT_INFILESIZE` do it for us already?

Reply:
To be honest, it's an unnecessary modify, if the send_length has specified correctly, it will not use `chunked encoding`